src/main/java/net/dv8tion/jda/client/entities/Application.java

```diff
@@ -0,0 +1,295 @@
+
+    /**
+     * Whether the bot requires code grant to invite or not. 
+     * If your application requires multiple scopes then you may need the full OAuth2 flow to ensure a
+     * bot doesn't join before your application is granted a token.
+     * 
+     * @return whether the bot requires code grant
+     */
+    boolean doesBotRequireCodeGrant();
+
+    /**
+     * Returns the Bot assigned to this Application
+     * 
+     * @return The {@link Application.Bot} assigned to this application, or null if no bot is assigned
+     */
+    Application.Bot getBot();
+
+    /**
+     * Returns the description of the application.
+     * 
+     * @return The description of the application or an empty {@link String} if no description is defined.
+     */
+    String getDescription();
+
+    /**
+     * Returns the flags for this application. It's used for whitelisted apps.
+     * 
+     * @return The application flags.
+     */
+    int getFlags();
+
+    /**
+     * Returns the icon id of the application.
+     * 
+     * @return The iconId of the application or null if no icon is defined.
+     */
+    String getIconId();
+
+    /**
+     * Returns the icon-url of the application.
+     * 
+     * @return The icon-url of the application or null if no icon is defined.
+     */
+    String getIconUrl();
+
+    /**
+     * The {@link net.dv8tion.jda.core.JDA JDA} instance of this Application
+     * (the one owning this application).
+     * 
+     * @return The JDA instance of this Application.
+     */
+    JDA getJDA();
+
+    /**
+     * Returns the {@link net.dv8tion.jda.client.managers.ApplicationManager ApplicationManager} for this Application.
+     * In the ApplicationManager, you can modify things like the name and icon of this Application.
+     *
+     * @return The ApplicationManager of this Channel
+     *
+     * @see    #getManagerUpdatable()
+     */
+    ApplicationManager getManager();
+
+    /**
+     * Returns the {@link net.dv8tion.jda.client.managers.ApplicationManagerUpdatable ApplicationManagerUpdatable} for this Application.
+     * In the ApplicationManager, you can modify things like the name and icon of this Application.
+     * <br>This can be used to bulk update application settings.
+     *
+     * @return The ApplicationManagerUpdatable of this Application
+     *
+     * @see    #getManager()
+     */
+    ApplicationManagerUpdatable getManagerUpdatable();
+
+    /**
+     * Returns the name of the application.
+     * 
+     * @return The name of the application.
+     */
+    String getName();
+
+    /**
+     * The redirect uris of this application.
+     * 
+     * @return A {@link java.util.List List} of current redirect uris of the application.
+     */
+    List<String> getRedirectUris();
+
+    /**
+     * The rpc application state of this application.
+     * 
+     * @return The rpc application state of current redirect uris of the application.
+     */
+    int getRpcApplicationState();
```

Comment:
The problem with that is: this isn't part of the documanted api and returns `0` for all of my apps